engines/producers/app/services/producers/product_finder.rb

```diff
@@ -6,7 +6,7 @@ class ProductFinder
     # @param producer [Producers::Producer]
     # @return [ActiveRecord::Relation<Producers::Product>]
     def find_by_producer(producer)
-      Product.where(producer: producer)
+      Product.where(producer_id: producer.id)
```

Comment:
I believe it's indifferent whether you use the id or not. What's the reason for the change?